src/integration/test_suites/smoke_test_mcu_mbox_ecc/smoke_test_mcu_mbox_ecc.c

```diff
@@ -0,0 +1,216 @@
+        }
+        clear_mcu_mbox_clear_sb_ecc_interrupt(mbox_num);
+    }
+
+    // Check if SB ECC status has been logged
+    if (mcu_mbox_read_hw_status(mbox_num) & MCU_MBOX0_CSR_MBOX_HW_STATUS_ECC_SINGLE_ERROR_MASK) {
+        VPRINTF(LOW, "MCU: Mbox%x SB ECC detected\n", mbox_num);
+    } else {
+        VPRINTF(FATAL, "MCU: Mbox%x No SB ECC detected\n", mbox_num);
+        SEND_STDOUT_CTRL(0x1);
+        while(1);
+    }
+
+    // Clear execute
+    mcu_mbox_clear_execute(mbox_num);
+    
+    // Check that ECC status registers are cleared
+    if (mcu_mbox_read_hw_status(mbox_num) & MCU_MBOX0_CSR_MBOX_HW_STATUS_ECC_SINGLE_ERROR_MASK) {
+        VPRINTF(FATAL, "MCU: Mbox%x SB ECC status not cleared\n", mbox_num);
+        SEND_STDOUT_CTRL(0x1);
+        while(1);
+    }
+
+    // MBOX: Enable ECC double bit injection
+    SEND_STDOUT_CTRL(TB_CMD_DISABLE_MBOX_SRAM_ECC_ERROR_INJECTION);
+    SEND_STDOUT_CTRL(TB_CMD_INJECT_MBOX_SRAM_DOUBLE_ECC_ERROR);
+ 
+    // MBOX: Acquire lock
+    if (!mcu_mbox_acquire_lock(mbox_num, 1000)) {
+        VPRINTF(FATAL, "MCU: Mbox%x didn't acquire lock\n", mbox_num);
+        SEND_STDOUT_CTRL(0x1);
+        while(1);        
+    }
+
+    // Single write to SRAM (Due to timing, ECC injector won't inject error on first write)
+    mcu_mbox_write_sram_dword(mbox_num, xorshift32() % (sram_size_kb * 256), xorshift32());
+
+    // Write random SRAM data to random addreses and read them back
+    for (uint32_t i = 0; i < (sizeof(mbox_addr)/sizeof(uint32_t)); i++) {
+        rand_data = xorshift32();
+        rand_dword_addr = xorshift32() % (sram_size_kb * 256);
+        mbox_addr[i] = rand_dword_addr;
+        mcu_mbox_write_sram_dword(mbox_num, rand_dword_addr, rand_data);
+    }
+
+    // Write short Dlen for test run time
+    mcu_mbox_write_dlen(mbox_num, 100);
+
+    // Setting execute (ECC reads are only checked during execute/transfer sessions)
+    mcu_mbox_set_execute(mbox_num);
+
+    for (uint32_t i = 0; i < (sizeof(mbox_addr)/sizeof(uint32_t)); i++) {
+        rand_dword_addr = mbox_addr[i];
+        rand_data = mcu_mbox_read_sram_dword(mbox_num, rand_dword_addr);
+
+        // Check if DB ECC interrupt has been logged
+        // R1WC to clear interrupt
+        if (!is_mcu_mbox_db_ecc_interrupt_set(mbox_num)) {
+            VPRINTF(FATAL, "MCU: Mbox%x DB ECC interrupt not set\n", mbox_num);
+            SEND_STDOUT_CTRL(0x1);
+            while(1);
+        }
+        clear_mcu_mbox_clear_db_ecc_interrupt(mbox_num);
+    }
+
+    // Check if DB ECC status has been logged
+    if (mcu_mbox_read_hw_status(mbox_num) & MCU_MBOX0_CSR_MBOX_HW_STATUS_ECC_DOUBLE_ERROR_MASK) {
+        VPRINTF(LOW, "MCU: Mbox%x DB ECC detected\n", mbox_num);
+    } else {
+        VPRINTF(FATAL, "MCU: Mbox%x No DB ECC detected\n", mbox_num);
+        SEND_STDOUT_CTRL(0x1);
+        while(1);
+    }
+
+    // Check HW_ERROR_NON_FATAL register
+    if (lsu_read_32(SOC_MCI_TOP_MCI_REG_HW_ERROR_NON_FATAL) & 
```

Comment:
Could you check that only ECC UCN is set? That way we guarantee this wire isn't tied to some other register. Can we do that in most places you are checking the interrupt bits? There was a lot of manual coding and is prone to error...